lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -87,45 +88,48 @@ public void OnNext(IDriverStarted value)
 
             public void OnNext(IAllocatedEvaluator value)
```

Comment:
I mainly changed it to use `StartTask` instead of `StartContext` because the check assumes the existence of a `FailedTask`. If the poison triggers before the Task starts, there is no guarantee that the Task is ever recorded.